src/cfnlint/decode/__init__.py

```diff
@@ -18,7 +18,7 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def decode(filename, ignore_bad_template):
+def decode(filename):
```

Comment:
Only using ignore_bad_template once and so many other issues with a template didn't handle this parameter.  Decided to remove it from here and move that logic outside of the decode.